backend/pkg/transport/network/tcp/tcp_test.go

```diff
@@ -0,0 +1,86 @@
+package tcp_test
+
+import (
+	"encoding/json"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/network/tcp"
+	"github.com/rs/zerolog"
+	"os"
+	"testing"
+	"time"
+)
+
+type Socket struct {
+	SrcIP   string
+	SrcPort uint16
+	DstIP   string
+	DstPort uint16
+}
+
+// Payload defines a piece of information comming from the network with its metadata
+type Payload struct {
+	Timestamp time.Time
+	Socket    Socket
+	Data      []byte
+}
+
+func TestTCP(t *testing.T) {
+	writer := os.Stdout
+	logger := zerolog.New(writer)
+
+	// Server setup
+	serverAddr := "127.0.0.1:3000"
```

Comment:
try to get the port some other way, there should be a way to get a port assigned by the os, that way it is sure that the test will run, what happens if the port 3000 is already in use. Search on google for how it is done in go